tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/bulkhead/clientserver/Checker.java

```diff
@@ -82,41 +85,38 @@ public Checker(int sleepMillis, TestData td, int fails) {
             while ((now > max) && !td.getMaxSimultaneousWorkers().compareAndSet(max, now)) {
                 max = td.getMaxSimultaneousWorkers().get();
             }
+
             if (fails > 0) {
-                Thread.sleep(millis / 2);
                 fails--;
-                RuntimeErrorException e = new RuntimeErrorException(new Error("fake error for Retry Testing"));
-                Utils.log(e.toString());
+                await().atMost(millis / 2, MILLISECONDS).
+                    untilAsserted(() -> assertEquals(fails, fails - 1));
```

Comment:
This looks wrong. When can `fails == fails - 1`?

Reply:
I don't think this was properly addressed. `await().atMost(TIME).untilAsserted(something that is never true)` is IMHO a very poor replacement for `Thread.sleep(TIME)`.

Reply:
actually here is all the opposite always is true because and putting this after decreasing the counter so we are ensuring that always will wait, but if you think that is so poor what do you suggest? thank you

Reply:
I think `assertEquals(fails, fails - 1)` can never be true (unless `fails` is a floating point number, then there are options :-) ).

Frankly, if we need to wait for certain amount of time, then `Thread.sleep` is best. `await().atMost(TIME).untilAsserted(() -> Assert.fail())` is exactly the same as `Thread.sleep(TIME)` (modulo interruptions), but it's IMHO much harder to read.

Reply:
well, I don't understand why you say that is not true the condition, the fail variable is an integer and after decreasing the value `fails --` I'm checking if the variable was decreased by one so always will wait `millis / 2` or maybe I can change the assertion for an `asserTrue` to make it clear. if is not enough for you I can let the Thread wait like was before.

Reply:
But you're doing `assertEquals(fails, fails - 1)`. That can't possibly be true. You'd have to store the old value into a different variable to be able to verify that `fails--` really decrements by 1 -- but I don't think such assert would make a lot of sense, we're not in the business of testing how Java operators work :-)

Reply:
ok i'll revert the change here. thank you

Reply:
Thanks!

Reply:
done